### PR TITLE
Update capi tag to v1.0.1 and port patches

### PR DESCRIPTION
**Pull request by mrajashree**
*Issue #, if available:* https://github.com/aws/eks-anywhere-build-tooling/issues/173

*Description of changes:*
- Update CAPI to v1.0.1 tag
- Update [capi golang version to 1.16](https://github.com/kubernetes-sigs/cluster-api/blob/v1.0.1/go.mod#L3)
- Changes to patches:
  - Ported over all patches for BR, external etcd, proxy and registry mirror changes. These patches are for all commits cherry-picked from capi fork's branch-0.3.19, and added to the new [capi fork rebased on capi v1.0.1 tag](https://github.com/mrajashree/cluster-api/tree/branch-1.0.1)
  - Removing the following patches because the changes are merged upstream: 
    - https://github.com/aws/eks-anywhere-build-tooling/blob/main/projects/kubernetes-sigs/cluster-api/patches/0001-update-coredns-corefile-migration-to-support-1.8.3.patch
    - https://github.com/aws/eks-anywhere-build-tooling/blob/main/projects/kubernetes-sigs/cluster-api/patches/0027-capd-do-not-reconcile-machine-if-cluster-or-machine-is-paused.patch
    - https://github.com/aws/eks-anywhere-build-tooling/blob/main/projects/kubernetes-sigs/cluster-api/patches/0030-Ignore-EndpointSlice-when-checking-for-cert-manager-.patch
   

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jaxesn** [comment]
Same comment, lets add an updating section to the readme.

File: projects/kubernetes-sigs/cluster-api/build/create_manifests.sh [hunk 1]
```diff
@@ -34,41 +34,28 @@ KUBE_RBAC_PROXY_LATEST_TAG=latest
...
+make set-manifest-pull-policy PULL_POLICY=IfNotPresent TARGET_RESOURCE="./bootstrap/kubeadm/config/default/manager_pull_policy.yaml"
+make set-manifest-pull-policy PULL_POLICY=IfNotPresent TARGET_RESOURCE="./controlplane/kubeadm/config/default/manager_pull_policy.yaml"
 
 ## Build the manifests
 make release-manifests
 
 ## Build the development manifests
 make -C test/infrastructure/docker set-manifest-image \
 MANIFEST_IMG=$CAPI_REGISTRY_PREFIX/capd-manager MANIFEST_TAG=$IMAGE_TAG
-sed -i 's,image: .*,image: '"${IMAGE_REPO}/brancz/kube-rbac-proxy:${KUBE_RBAC_PROXY_LATEST_TAG}"',' ./test/infrastructure/docker/config/manager/manager_auth_proxy_patch.yaml
```
Comment (jaxesn) [inline projects/kubernetes-sigs/cluster-api/build/create_manifests.sh:71]: did they pull kube-rbac-proxy from capd?

File: projects/kubernetes-sigs/cluster-api/README.md [hunk 2]
```diff
@@ -11,3 +11,25 @@ You can find the latest versions of these images on ECR Public Gallery.
...
+   1. Review the patches under patches/ folder in this repo. Apply the required patches to the new branch created in the above step. Remove any patches that are either
+   merged upstream or no longer needed. Please reach out to @jgw, @vgg, @gaslor or @mrajashree if there are any questions regarding keeping/removing patches.
+   1. Run `git format-patch <commit>`, where `<commit>` is the last upstream commit on that tag. Move the generated patches under the patches/ folder in this repo.
+1. Update the `GIT_TAG` file to have the new desired version based on the upstream release tags.
+1. Compare the old tag to the new, looking specifically for Makefile changes.
+   ex: [0.3.19 compared to 1.0.1](https://github.com/kubernetes-sigs/cluster-api/compare/v0.3.19...v1.0.1). Check the targets in the Makefile
+   for any changes.
+   1. For instance, [the make targets for creating manifests in v1.0.1 use a different path for manager_image_patch.yaml as compared to v0.3.19](https://github.com/kubernetes-sigs/cluster-api/commit/280db9a796d5e1c2b3b75aa3036fcfe44f669909#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L368-L375).
```
Comment (jaxesn) [inline projects/kubernetes-sigs/cluster-api/README.md:29]: 💰

File: projects/kubernetes-sigs/cluster-api/README.md [hunk 3]
```diff
@@ -11,3 +11,25 @@ You can find the latest versions of these images on ECR Public Gallery.
...
+
+### Updating
+
+1. Review releases and changelogs in upstream [repo](https://github.com/kubernetes-sigs/cluster-api) and decide on new version.
+   Please review carefully and if there are questions about changes necessary to eks-anywhere to support the new version
+   and/or automatically update between eks-anywhere version reach out to @jgw, @vgg, @gaslor or @mrajashree.
+1. Follow these steps for changes to the patches/ folder:
+   1. Checkout the desired tag on our [CAPI fork](https://github.com/mrajashree/cluster-api) and create a new branch.
+   1. Review the patches under patches/ folder in this repo. Apply the required patches to the new branch created in the above step. Remove any patches that are either
+   merged upstream or no longer needed. Please reach out to @jgw, @vgg, @gaslor or @mrajashree if there are any questions regarding keeping/removing patches.
```
Comment (jaxesn) [inline projects/kubernetes-sigs/cluster-api/README.md:23]: real usernames?

**Review by jaxesn — CHANGES_REQUESTED** [review]
nits

File: projects/kubernetes-sigs/cluster-api/build/create_manifests.sh (hunk 1 above)
Comment (mrajashree) [reply projects/kubernetes-sigs/cluster-api/build/create_manifests.sh:71]: yup, as a part of this PR https://github.com/kubernetes-sigs/cluster-api/pull/4640/files

**Review by jaxesn — APPROVED** [review]
/lgtm

**Pull request merged**